app/src/main/java/com/teamagam/gimelgimel/app/map/viewModel/SendGeometryViewModel.java

```diff
@@ -53,5 +93,82 @@ public String getDescription() {
   public void onLocationSelection(PointGeometry point) {
     sLogger.userInteraction(
         "Send geometry location selected " + point.getLatitude() + "/" + point.getLongitude());
+    mSelectedPoints.add(point);
+    refreshDisplayedGeometry();
+  }
+
+  private boolean isValidForm() {
+    return !isEmpty(mDescription) && isValidGeometry();
+  }
+
+  private boolean isValidGeometry() {
+    if (isPolylineState()) {
+      return mSelectedPoints.size() > 1;
+    } else {
+      return mSelectedPoints.size() > 2;
+    }
+  }
+
+  private Geometry getCurrentGeometry() {
+    if (isPolylineState()) {
+      return new Polyline(mSelectedPoints);
+    }
+    return new Polygon(mSelectedPoints);
+  }
+
+  private void removeLastSelectedPoint() {
+    mSelectedPoints.remove(mSelectedPoints.size() - 1);
+  }
+
+  private void refreshDisplayedGeometry() {
+    removeOldDisplayedGeometry();
+    displayCurrentGeometry();
+  }
+
+  private void removeOldDisplayedGeometry() {
+    mMapDrawer.clear();
+  }
+
+  private void displayCurrentGeometry() {
+    switch (mSelectedPoints.size()) {
+      case 0:
+        break;
+      case 1:
+        displayPoint(mSelectedPoints.get(0));
+        break;
+      case 2:
+        displayPolyline(mSelectedPoints);
+        break;
+      default:
+        displayBySwitch();
+    }
+  }
+
+  private void displayPoint(PointGeometry point) {
+    mMapDrawer.draw(mMapEntityFactory.createPoint(point));
+  }
+
+  private void displayPolyline(List<PointGeometry> points) {
+    mMapDrawer.draw(mMapEntityFactory.createPolyline(points));
+  }
+
+  private void displayBySwitch() {
+    if (isPolylineState()) {
+      displayPolyline(mSelectedPoints);
+    } else {
+      displayPolygon(mSelectedPoints);
+    }
+  }
+
+  private boolean isPolylineState() {
+    return !mIsSwitchChecked;
+  }
+
+  private void displayPolygon(List<PointGeometry> points) {
+    mMapDrawer.draw(mMapEntityFactory.createPolygon(points));
+  }
+
+  public interface InvalidInputNotifier {
```

Comment:
Isn't it a common behavior that SendQuad... and SendGeo... should share? I mean, the validation process- I think it belongs on both.